crates/oxc_linter/src/rules/react/jsx_no_script_url.rs

```diff
@@ -0,0 +1,272 @@
+}
+
+const IS_JAVA_SCRIPT_PROTOCOL: &str = r"(j|J)[\r\n\t]*(a|A)[\r\n\t]*(v|V)[\r\n\t]*(a|A)[\r\n\t]*(s|S)[\r\n\t]*(c|C)[\r\n\t]*(r|R)[\r\n\t]*(i|I)[\r\n\t]*(p|P)[\r\n\t]*(t|T)[\r\n\t]*:";
+
+#[derive(Debug, Default, Clone)]
+pub struct JsxNoScriptUrl(Box<JsxNoScriptUrlConfig>);
+
+#[derive(Debug, Default, Clone)]
+pub struct JsxNoScriptUrlConfig {
+    include_from_settings: bool,
+    components: FxHashMap<String, Vec<String>>,
+}
+
+impl std::ops::Deref for JsxNoScriptUrl {
+    type Target = JsxNoScriptUrlConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow usage of `javascript:` URLs
+    ///
+    /// ### Why is this bad?
+    ///
+    /// URLs starting with javascript: are a dangerous attack surface because it’s easy to accidentally include unsanitized output in a tag like <a href> and create a security hole.
+    /// In React 16.9 any URLs starting with javascript: scheme log a warning.
+    /// In a future major release, React will throw an error if it encounters a javascript: URL.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```jsx
+    /// <a href="javascript:void(0)">Test</a>
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```jsx
+    /// <Foo test="javascript:void(0)" />
+    /// ```
+    JsxNoScriptUrl,
+    suspicious,
+    pending
+);
+
+fn check_is_link_attribute(tag_name: &str, prop_value_literal: String, ctx: &LintContext) -> bool {
+    tag_name == "a"
+        || ctx.settings().react.get_link_component_attrs(tag_name).is_some_and(
+            |link_component_attrs| {
+                link_component_attrs.contains(&CompactStr::from(prop_value_literal))
+            },
+        )
+}
+
+impl JsxNoScriptUrl {
+    fn check_is_link(&self, tag_name: &str, ctx: &LintContext) -> bool {
+        if !self.include_from_settings {
+            return tag_name == "a";
+        }
+        if tag_name == "a" {
+            return true;
+        }
+        ctx.settings().react.get_link_component_attrs(tag_name).is_some()
+    }
+}
+
+impl Rule for JsxNoScriptUrl {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::JSXOpeningElement(element) = node.kind() {
+            let Some(component_name) = element.name.get_identifier_name() else {
+                return;
+            };
+            if let Some(link_props) = self.components.get(component_name.as_str()) {
+                for jsx_attribute in &element.attributes {
+                    if let JSXAttributeItem::Attribute(attr) = jsx_attribute {
+                        let Some(prop_value) = &attr.value else {
+                            return;
+                        };
+                        if prop_value.as_string_literal().is_some_and(|val| {
+                            let re = Regex::new(IS_JAVA_SCRIPT_PROTOCOL).unwrap();
```

Comment:
you should use `lazy_static` to wrap it to make sure it is only initialized once. https://github.com/oxc-project/oxc/blob/a786dc2449c46112cd3a5d3f99ea6087e39c3075/crates/oxc_linter/src/rules/eslint/no_nonoctal_decimal_escape.rs#L87-L90

Reply:
here is the reason, https://github.com/rust-lang/regex?tab=readme-ov-file#usage-avoid-compiling-the-same-regex-in-a-loop, I recommended using `lazy_static` since it is widely used in `linter` crate